monkey/monkey_island/cc/environment/environment_config.py

```diff
@@ -35,31 +33,29 @@ def _load_from_file(self, file_path):
 
     def _load_from_json(self, config_json: str) -> EnvironmentConfig:
         data = json.loads(config_json)
-        self._load_from_dict(data)
+        self._load_from_dict(data["environment"])
 
     def _load_from_dict(self, dict_data: Dict):
         aws = dict_data["aws"] if "aws" in dict_data else None
-        data_dir = dict_data["data_dir"] if "data_dir" in dict_data else DEFAULT_DATA_DIR
 
         self.server_config = dict_data["server_config"]
         self.deployment = dict_data["deployment"]
         self.user_creds = _get_user_credentials_from_config(dict_data)
         self.aws = aws
-        self.data_dir = data_dir
-
-    @property
-    def data_dir_abs_path(self):
-        return os.path.abspath(os.path.expanduser(os.path.expandvars(self.data_dir)))
 
     def save_to_file(self):
+        with open(self._server_config_path, "r") as f:
+            config = json.load(f)
+
+        config["environment"] = self.to_dict()
+
         with open(self._server_config_path, "w") as f:
-            f.write(json.dumps(self.to_dict(), indent=2))
+            f.write(json.dumps(config, indent=2))
 
     def to_dict(self) -> Dict:
```

Comment:
I think we should rename this to something that reflects that it's only converting the "environment" part of the server config to a dict. Similarly for `_load_from_dict()`; maybe `_load_env_config_from_dict()`?

Reply:
An example usage would be something like
```
environment_config = EnvironmentConfig(filename)
environment_config.save_to_file()
```

Do you think it needs to be more explicit?

As a side note, this class really shouldn't be modifying the config. This is a symptom of a larger issues, which is that there is config data entangled with state data. The username and password hash are really elements of state, not elements of configuration. A lot of this logic will go away if/when we resolve this problem. At that time, we can pass only the appropriate `dict` object to this class, rather than a file path it needs to write to.

Reply:
Sorry, I wasn't clear. I was referring to `to_dict()`. It only converts whatever is under "environment" in the server config. I think my main concern is that the class is called `EnvironmentConfig` and parts of it (like `to_dict()`) handle only the "environment" part of the server config, but certain other parts (like `_load_from_file()`) are working with the whole server config, which is confusing.

> As a side note, this class really shouldn't be modifying the config. This is a symptom of a larger issues, which is that there is config data entangled with state data. The username and password hash are really elements of state, not elements of configuration. A lot of this logic will go away if/when we resolve this problem. At that time, we can pass only the appropriate dict object to this class, rather than a file path it needs to write to.

But agreed. Maybe we should change the function names for now so it's easier to understand what's going on in case we don't get to this soon.

Reply:
I think that since `to_dict()` operates on an EnvironmentConfig object, like `environment_config.to_dict()`, this indicates that it converts an environment config to a dict. This is the nature of object-oriented programming. I think naming the function `to_environment_config_dict()` would be redundant.

Reply:
(Discussed in a meeting and agreed to leave this as is until we decide to change/untangle the structure.)